src/wrappers/AppInitializer.tsx

```diff
@@ -1,30 +1,48 @@
 import { useEffect, useState } from 'react';
 import { Address } from '@elrondnetwork/erdjs/out';
 import { useGetAccountInfo } from 'hooks';
+import { loginAction } from 'redux/commonActions';
 import { useDispatch } from 'redux/DappProviderContext';
+import { initializeExtraActions } from 'redux/slices/extraActionsSlice';
 import { initializeNetworkConfig } from 'redux/slices/networkConfigSlice';
-import { NetworkConfigType } from 'types';
+import { ExtraActionsType, NetworkConfigType } from 'types';
 import { logout } from 'utils';
 import getAccountShard from 'utils/account/getAccountShard';
 
 export function AppInitializer({
   networkConfig,
+  extraActions,
   children
 }: {
   networkConfig: NetworkConfigType;
   children: any;
+  extraActions?: ExtraActionsType;
 }) {
   const [initialized, setInitialized] = useState(false);
   const account = useGetAccountInfo();
   const { address, publicKey } = account;
   const dispatch = useDispatch();
 
-  async function initializeApp() {
+  useEffect(() => {
     dispatch(initializeNetworkConfig(networkConfig));
-    setInitialized(true);
+    if (extraActions !== undefined) {
+      extraActions.init({
```

Comment:
I think it would be a good idea to use the optional chaining to avoid crashing the app. Maybe also wrap all this in a try catch to avoid crashing the whole app if this initialization fails

Reply:
OK - added optional chaining.
OnLogin is strongly typed so I guess no errors can occur there. But we can add try catch blocks if we identify possible error places

Reply:
OK